packages/server/lib/browsers/webdriver-classic/index.ts

```diff
@@ -0,0 +1,272 @@
+
+        try {
+          const resp = await createSessionResp.json()
+
+          error.message = `${error.message } ${resp.value.error}. ${resp.value.message}.`
+        } finally {
+          // if for some reason we can't parse the response, continue to throw with some information.
+          throw error
+        }
+      }
+
+      const createSessionRespBody = await createSessionResp.json()
+
+      this.sessionId = createSessionRespBody.value.sessionId
+
+      return createSessionRespBody.value
+    } catch (e) {
+      debug(`unable to create new Webdriver session: ${e}`)
+      throw e
+    }
+  }
+
+  /**
+   * Gets available windows handles in the browser. The order in which the window handles are returned is arbitrary.
+   * @see https://www.w3.org/TR/webdriver2/#get-window-handles
+   *
+   * @returns {Promise<string[]>} All the available top-level contexts/handles
+   */
+  async getWindowHandles (): Promise<string[]> {
+    const getWindowHandles = `http://${this.#host}:${this.#port}/session/${this.sessionId}/window/handles`
+
+    try {
+      const getWindowHandlesResp = await crossFetch(getWindowHandles)
+
+      if (!getWindowHandlesResp.ok) {
+        throw new Error(`${getWindowHandlesResp.status}: ${getWindowHandlesResp.statusText}`)
+      }
+
+      const getWindowHandlesRespBody = await getWindowHandlesResp.json()
+
+      return getWindowHandlesRespBody.value
+    } catch (e) {
+      debug(`unable to get classic webdriver window handles: ${e}`)
+      throw e
+    }
+  }
+
+  /**
+   * Switching window will select session's current top-level browsing context used as the target for all subsequent commands.
+   * In a tabbed browser, this will typically make the tab containing the browsing context the selected tab.
+   * @see https://www.w3.org/TR/webdriver2/#dfn-switch-to-window
+   *
+   * @param {string} handle - the context ID of the window handle
+   * @returns {Promise<null>}
+   */
+  async switchToWindow (handle: string): Promise<null> {
+    const switchToWindowUrl = `http://${this.#host}:${this.#port}/session/${this.sessionId}/window`
+
+    const headers = new Headers()
+
+    headers.append('Content-Type', 'application/json')
+
+    const body = {
+      handle,
+    }
+
+    try {
+      const switchToWindowResp = await crossFetch(switchToWindowUrl, {
+        method: 'POST',
+        headers,
+        body: JSON.stringify(body),
+      })
+
+      if (!switchToWindowResp.ok) {
+        throw new Error(`${switchToWindowResp.status}: ${switchToWindowResp.statusText}`)
+      }
+
+      const switchToWindowRespBody = await switchToWindowResp.json()
+
+      return switchToWindowRespBody.value
+    } catch (e) {
+      debug(`unable to switch to window via classic webdriver : ${e}`)
+      throw e
+    }
+  }
+
+  /**
+   * maximizes the current window
+   * @see https://www.w3.org/TR/webdriver2/#maximize-window
+   *
+   * @returns {Promise<null>}
+   */
+  async maximizeWindow (): Promise<null> {
```

Comment:
A lot of this code seems similar from method to method. Wondering if we can DRY it up a bit.

Reply:
probably. I don't think its worth the effort though since this code is going to be removed in the near future